### DISCUSSION TITLE
mini PCI coral wont work anymore

### DISCUSSION THREAD

**Discussion by matteopizzoli** [blakeblackshear/frigate#3509] [General]
Hi,
i'm rewriting all the configuration with the new installation, but frigate won't start :
Attaching to frigate
frigate  | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate  | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate  | [fix-attrs.d] applying ownership & permissions fixes...
frigate  | [fix-attrs.d] done.
frigate  | [cont-init.d] executing container initialization scripts...
frigate  | [cont-init.d] done.
frigate  | [services.d] starting services
frigate  | [services.d] done.
frigate  | [2022-07-23 00:50:23] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
frigate  | Starting migrations
frigate  | [2022-07-23 00:50:24] peewee_migrate                 INFO    : Starting migrations
frigate  | There is nothing to migrate
frigate  | [2022-07-23 00:50:24] peewee_migrate                 INFO    : There is nothing to migrate
frigate  | [2022-07-23 00:50:24] frigate.app                    INFO    : Output process started: 219
frigate  | [2022-07-23 00:50:24] detector.coral1                INFO    : Starting detection process: 217
frigate  | [2022-07-23 00:50:24] frigate.edgetpu                INFO    : Attempting to load TPU as pci:0
frigate  | [2022-07-23 00:50:24] frigate.edgetpu                ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet                                                                                                     , you must configure CPU detectors.
frigate  | Process detector:coral1:
frigate  | Traceback (most recent call last):
frigate  |   File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate
frigate  |     delegate = Delegate(library, options)
frigate  |   File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__
frigate  |     raise ValueError(capture.message)
frigate  | ValueError
frigate  |
frigate  | During handling of the above exception, another exception occurred:
frigate  |
frigate  | Traceback (most recent call last):
frigate  |   File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
frigate  |     self.run()
frigate  |   File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
frigate  |     self._target(*self._args, **self._kwargs)
frigate  |   File "/opt/frigate/frigate/edgetpu.py", line 136, in run_detector
frigate  |     object_detector = LocalObjectDetector(
frigate  |   File "/opt/frigate/frigate/edgetpu.py", line 44, in __init__
frigate  |     edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config)
frigate  |   File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 162, in load_delegate
frigate  |     raise ValueError('Failed to load delegate from {}\n{}'.format(
frigate  | ValueError: Failed to load delegate from libedgetpu.so.1.0
frigate  |
frigate  | [2022-07-23 00:50:24] frigate.app                    INFO    : Camera processor started for uscitaanteriore: 224
frigate  | [2022-07-23 00:50:24] ws4py                          INFO    : Using epoll
frigate  | [2022-07-23 00:50:24] frigate.app                    INFO    : Capture process started for uscitaanteriore: 225
frigate  | [2022-07-23 00:50:24] ws4py                          INFO    : Using epoll
frigate  | [2022-07-23 00:50:44] frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
frigate  | [cont-finish.d] executing container finish scripts...
frigate  | [cont-finish.d] done.
frigate  | [s6-finish] waiting for services.
frigate  | [s6-finish] sending all processes the TERM signal.
frigate  | [s6-finish] sending all processes the KILL signal and exiting.
frigate exited with code 0
ffrigate  | [2022-07-23 00:51:09] frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
frigate  | [cont-finish.d] executing container finish scripts...
frigate  | [cont-finish.d] done.
frigate  | [s6-finish] waiting for services.
frigate  | [s6-finish] sending all processes the TERM signal.
^CGracefully stopping... (press Ctrl+C again to force)

the problem is the detector if i omit this part everityng run but obiously more slow (inference time 72 vs old 10) but stable.
i have a minipci e detector running on docker in ubuntu 22.04.
This work perfect since last version.
the config is:
...
mqtt:
  host: 192.168.1.3
  port: 1883
  topic_prefix: frigate
  user: mqttuser
  password: xxx
detectors:
  coral1:
    type: edgetpu
    device: pci:0
cameras:
  uscitaanteriore:
   ffmpeg:
      hwaccel_args:
       - -hwaccel
       - vaapi
       - -hwaccel_device
       - /dev/dri/renderD128
       - -hwaccel_output_format
       - yuv420p
      inputs:
       - path: rtsp://192.168.1.128:554/user=admin_password=tlJwpbo6_channel=1_stream=1.sdp?real_stream
         roles:
           - detect
           - rtmp
   rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
   detect:
     height: 480
     width: 640
     fps: 10
...

even if i write:
...
detectors:
  coral:
    type: edgetpu
    device: pci
...

the frigate seem to start and in few online moments i can read on debug (see the image attached)..but quit after few moments.
i don't know if the edge tpu is really read or not
![debug](https://user-images.githubusercontent.com/17899780/180578743-11ab1872-3e95-4257-bfa9-f1c7b3ab875d.PNG)

**Comment by NickM-27** [comment]
10 is the default inference speed so that doesn't indicate much. Have you reinstalled the coral drivers? those are required for pci based corals to work

**Comment by matteopizzoli** [comment]
i did not remeber i have installed in last server...but i found this:
https://coral.ai/docs/m2/get-started/#2-install-the-pcie-driver-and-edge-tpu-runtime
it is correct for you?
thanks a lot!!!

**Reply by NickM-27** [reply]
That would be it

**Reply by matteopizzoli** [reply]
Solved thankyou!!
